fix: dispatch Jira events based on domain + webhook id

pashafateev opened the pull request:
### tl;dr:  
This update modifies how Jira events are dispatched by incorporating both the **Jira domain** and **webhook ID** when identifying connections.  

#### Key Changes:  
1. **Introduced `webhookKey`**:  
   - A new function `webhookKey(domain, webhookID string) string` creates a unique key by combining the Jira domain with the webhook ID.  

2. **Updated Webhook Key Handling**:  
   - Added a new `WebhookKeySymbol` in `vars.go` to represent this composite key.  

3. **Refactored Issue URL Handling**:  
   - Introduced `transformIssueURL` to normalize the Jira issue URL before constructing the webhook key.  

4. **Modified Event Dispatch Logic**:  
   - Instead of using `webhookID` alone, event dispatch now uses `webhookKey` for identifying connections (`FindConnectionIDs` now queries using `WebhookKeySymbol`).  

### Testing
Manually tested by initializing a new Jira connection and successfully dispatching events.

pashafateev committed:
dispatch Jira events based on domain + webhook id

pashafateev committed:
add transformIssueURL()

daabr commented on integrations/atlassian/jira/event.go:
```diff
@@ -106,10 +107,17 @@ func (h handler) handleEvent(w http.ResponseWriter, r *http.Request) {
 
+	u, err := transformIssueURL(jiraEvent, l)
+	if err != nil {
+		l.Error("Failed to transform issue URL", zap.Error(err))
```
1. Not all Jira events relate to issues
2. It's easier to understand the initial meaning/context of this log message in Datadog when you specify it's related to Jira
```suggestion
		l.Error("Failed to transform Jira URL", zap.Error(err))
```

daabr commented on integrations/atlassian/jira/event.go:
```diff
@@ -122,6 +130,30 @@ func (h handler) handleEvent(w http.ResponseWriter, r *http.Request) {
+// transformIssueURL is used to transform the issue URL in the Jira event
+// to a normalized URL that can be used for the webhook key.
+func transformIssueURL(jiraEvent map[string]any, l *zap.Logger) (string, error) {
+	issue, ok := jiraEvent["issue"].(map[string]any)
```
What if the event isn't related to issues?

pashafateev replied:
[These are the only supported events](https://developer.atlassian.com/cloud/jira/platform/webhooks/#registering-a-webhook-using-the-rest-api--for-connect-and-oauth-2-0-apps-
) for an OAuth 2.0 app. 

The request headers don't have the URL and the only option I see is using issue from the event

pashafateev committed:
Update integrations/atlassian/jira/event.go

Co-authored-by: Daniel Abraham <daniel@autokitteh.com>

pashafateev committed:
fix(integrations): validate before saving api key in chatgpt (#1105)

This bug was reported by @haimzlato, where running ChatGPT in production
with an invalid ID caused an infinite loop. I was unable to reproduce
this issue locally or on the latest production version. However, I did
address a related issue: the lack of API key validation.

**Testing**: Manually tested both success and failure cases locally.

itayd committed:
feat: add session_id to parameters when starting a session (#1107)

this can work now:

```
def on_trigger(event):
  print(event.session_id)
  print(event.data)
```

sometimes the user might want to correlate the session id with data that
is sent elsewhere. we might also can associate more data to the event
other than the input data.

daabr committed:
refactor(CI): improve build time & isolation (#1108)

Part 1 of refactoring CI: don't run any more or less, just segment it
differently, enable caching, and avoid duplication.

- Go:
- Don't use `make bin/ak` - it blocks caching (due to timestamp-based
build flags)
  - Split between unit tests and long-running system tests
- Don't run session tests in unit-tests job, already running in other
jobs
- Don't run `go mod download` - useful in the past, but not anymore with
better caching
- Python: move Python runtime's Go tests to Python workflow (due to
Python dependency)

Stats:

- Go in main: 12m32s
-
https://github.com/autokitteh/autokitteh/actions/runs/13466518794/job/37633353273
- Go in this PR: 2m52s + 7m42s = 10m34s
-
https://github.com/autokitteh/autokitteh/actions/runs/13477279622/job/37657993992?pr=1108
-
https://github.com/autokitteh/autokitteh/actions/runs/13477279622/job/37658023378?pr=1108

Refs: INT-2014

itayd committed:
fix: broken python test in main (#1112)

itayd committed:
chore: remove no longer needed integration funcs (#1109)

daabr committed:
refactor(CI): split OPA and Stalark (#1110)

Part 2 of refactoring CI: split OPA and Starklark test into their own
workflows.

This doesn't improve runtime or reliability, but it improves readability
of the Go workflow.

Personally I think we should get rid of the Starlark tests altogether,
but @itayd has plans to maybe migrate stuff from session tests?

Refs: INT-2014, stacked on top of #1108

daabr committed:
refactor(CI): session tests (#1111)

Fix 1: `make test-sessions` didn't depend on `bin/ak` - this is a
mistake because it requires it, and we can't just assume that it's
up-to-date.

But... Fix 2: this means we also need to build AK and run `run.sh`
directly in the Go and Python CI workflows, to make GitHub caching
effective like in #1108, and avoid building AK twice.

Refs: INT-2014

mario99logic committed:
feat: Add soft delete for org and block org deletion if it has associated projects (#1078)

Added soft deletion functionality for organizations and implements
safeguards around org deletion:

Key changes:
- Added `deleted_at` column to organizations table for soft delete 
- Prevents org deletion if it has associated projects
- Added row-level locking during project creation to prevent race
conditions
- Renames org to its ID before soft deletion

Technical details:
- Uses row-level `FOR UPDATE` lock when creating projects to prevent
race conditions during org deletion
- Added validation check `doesOrgHasProject` to ensure orgs with
projects cannot be deleted
- Migration added for `deleted_at` column

Testing:
- tried deleting an org with associated projects
- deleting an org with no projects
- checking the name of the org and deleted_at column in postgres after
deletion

tebeka committed:
bug(python): Fix "start" in another module. (#818)

Fixes: ENG-1749

Note: This currently doesn't solve the bug, see issue for more details.

daabr committed:
refactor(CI): explicit step for each DB type in DB tests (#1113)

GitHub doesn't recognize the similarity between the old `ci-db.yml` and
new `db.yml` file, so here it is manually:

```diff
< name: DBGorm tests (all supported engines)
---
> # https://github.com/marketplace/actions/setup-go-environment
>
> name: CI - Databases
```

Always run when merging to main:

```diff
<   push:
<     branches:
<       - main
<     paths:
<       - "internal/backend/db/dbgorm/**"
<   pull_request:
<     paths:
<       - "internal/backend/db/dbgorm/**"
---
>   pull_request:
>     paths:
>       - .github/workflows/db.yml
>       - internal/backend/db/**
>   push:
>     branches: [main]
```

No need for 10 minutes, this finishes in 1 minute:

```diff
<   validate:
<     name: DBGorm tests
<     timeout-minutes: 10
---
>   db-tests:
>     name: Database tests
>     timeout-minutes: 5
```

Explicit `go test` steps, instead of `make`:

```diff
<       - name: Run DBGorm tests on all engines
<         run: make test-dbgorm
---
>       - name: Install gotestsum
>         run: go install gotest.tools/gotestsum@latest
>
>       - name: Test PostgreSQL
>         run: gotestsum -f testname -- -trimpath ./internal/backend/db/... -dbtype postgres
>
>       # This is the default DB type, so it's already a part of the Go unit tests,
>       # but it doesn't hurt to run it again side-by-side with other database types.
>       - name: Test SQLite
>         run: gotestsum -f testname -- -trimpath ./internal/backend/db/... -dbtype sqlite
```

Refs: INT-2014

daabr committed:
refactor(CI): bring back "go mod download" (#1114)

It does have some potential time-saving impact in the first time you run
checks for a new PR: somewhere between 5 and 50 seconds.

Also update golangci-lint version on the way.

Ref: INT-2014, #1108

daabr committed:
refactor(CI): delete obsolete cmd & deterministic test names (#1115)

Some table tests use non-deterministic and/or very long names. This
impacts test caching in local reruns.

Also delete a very old and obsolete skeleton of a CLI helper on the way.

Refs: INT-2014, ENG-415

tebeka committed:
fix(python): Catch code load errors and send to session log (#1117)

Fixes: ENG-2027

tebeka committed:
bug(python): Catch BaseException instead of Exception (#1124)

Some function (sys.exit for example) raise exceptions that do not
inherit from Exception.

Fixes: ENG-2028

daabr committed:
refactor(CI): minor cosmetic facelift for remaining workflows (#1121)

Refs: ENG-2014

daabr committed:
refactor(CI): Docker images (#1120)

1. Split Docker jobs out of the Go workflow: they're not related to Go,
and we won't see an unexecuted job in every PR and merge-to-main now

2. Note that the job to publish the image when the PR is merged to main
wasn't triggered as expected: the `if` means that the job got triggered
when the `needs` jobs end, regardless of their success or failure.

   ```yaml
   needs:
     - go-unit-tests
     - go-session-tests
     - go-system-tests
     - static-analysis
   if: github.ref == 'refs/heads/main'
   ```

(If we want to block, we can use AutoKitteh - we have a sample exactly
for the scenario of
   triggering a workflow when a group of other workflows are successful)

3. Remove `with: fetch-depth: 0` from `checkout` - it means full
history, not just the latest commit (which is the default behavior) -
it's needed only in Go Releaser

4. `release.yml`: changed trigger from `push: tags: "v*"` to `release:
types: [published]` (see
https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#release)

5. `WORKING_DIRECTORY` removed: not needed since we use the root dir of
a monorepo

6. `${GITHUB_REF#refs/tags/}` --> `${GITHUB_REF_NAME}` (same thing,
without string substitution, which doesn't even work if the ref isn't a
tag)

7. Change timeout from 30 minutes to 10 (it always takes 5 minutes to
finish in all modes)

8. Misc version updates of GitHub Actions and cosmetic tweaks

Refs: ENG-2014

pashafateev committed:
simplify webhook key

pashafateev committed:
Merge branch 'main' into pasha/int-174
